### PR TITLE
[archer2] Add initial configuration

### LINKED ISSUE #43: Add Archer2 configuration

### PR DESCRIPTION
**Pull request by giordano**
This should be working, but I'm having lots of troubles with submitting the jobs.  I believe this is at least in part related to the fact ARCHER2 compute nodes can't access the home directory, but it isn't clear to me _what_ is trying to access the home directory, I'll need to investigate more.

Fix #43

### CODE REVIEW

**Comment by giordano** [comment]
This is part of the issue, but there is more to it: https://github.com/spack/spack/issues/33265.

**Comment by giordano** [comment]
With the last two commit, to work around the issues I mentioned above, I was able to successfully run the example benchmark on archer2:
```
[==========] Running 1 check(s)
[==========] Started on Thu Oct 13 10:55:33 2022

[----------] start processing checks
[ RUN      ] SombreroBenchmark /cc391b86 @archer2:compute-node+default
[       OK ] (1/1) SombreroBenchmark /cc391b86 @archer2:compute-node+default
P: flops: 1.16 Gflops/seconds (r:1, l:None, u:None)
[----------] all spawned checks have finished

[  PASSED  ] Ran 1/1 test case(s) from 1 check(s) (0 failure(s), 0 skipped)
[==========] Finished on Thu Oct 13 10:58:03 2022

============================================================================================================================================================================================================================================
PERFORMANCE REPORT
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
[SombreroBenchmark /cc391b86 @archer2:compute-node:default]
  num_tasks: 1
  num_gpus_per_node: 0
  num_tasks_per_node: 1
  num_cpus_per_task: 1
  performance:
    - flops: 1.16 Gflops/seconds (r: 1 Gflops/seconds l: -inf% u: +inf%)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

File: spack-environments/archer2/compute-node/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        cflags: null
+        cxxflags: null
+        fflags: null
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/8.0.0
+      environment: {}
+      extra_rpaths: []
```
Comment (aturner-epcc) [inline spack-environments/archer2/compute-node/spack.yaml:104]: I do not see this compiler installed on ARCHER2?

File: spack-environments/archer2/compute-node/spack.yaml [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+        cxxflags: null
+        fflags: null
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/8.1.0
+      environment: {}
+      extra_rpaths: []
+  - compiler:
```
Comment (aturner-epcc) [inline spack-environments/archer2/compute-node/spack.yaml:123]: Not available on ARCHER2

File: spack-environments/archer2/compute-node/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+        cflags: null
+        cxxflags: null
+        fflags: null
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-aocc
+      - aocc/8.3.3
+      environment: {}
+      extra_rpaths: []
```
Comment (aturner-epcc) [inline spack-environments/archer2/compute-node/spack.yaml:140]: Not available on ARCHER2

File: spack-environments/archer2/compute-node/spack.yaml [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-gnu
+      - gcc/8.1.0
+      environment: {}
+      extra_rpaths: []
```
Comment (aturner-epcc) [inline spack-environments/archer2/compute-node/spack.yaml:200]: This compiler should not be included - GCC 8 is only installed to support CCE, it does not work as a standalone compiler environment

File: spack-environments/archer2/compute-node/spack.yaml [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-gnu
+      - gcc/10.2.0
+      environment: {}
+      extra_rpaths: []
```
Comment (aturner-epcc) [inline spack-environments/archer2/compute-node/spack.yaml:230]: For GCC >=10, you will probably need to add:

```
fflags: -fallow-argument-mismatch
```

To allow MPI code to compile

File: spack-environments/archer2/compute-node/spack.yaml [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+        f77: ftn
+        fc: ftn
+      flags: {}
+      operating_system: sles15
+      target: any
+      modules:
+      - PrgEnv-amd
+      - rocmcc/2.14
+      environment: {}
+      extra_rpaths: []
```
Comment (aturner-epcc) [inline spack-environments/archer2/compute-node/spack.yaml:290]: These compilers are not available on ARCHER2

**Review by aturner-epcc — CHANGES_REQUESTED** [review]
Just a few notes on the compiler environments available and additional options to support Fortran + MPI compilation with GCC versions >= 10

**Comment by giordano** [comment]
Interesting, I've got all compilers with `spack compiler find` on a login node.  Anyway, I removed the extra compilers and set `fflags` for GCC >= 10.

**Review by tkoskela — APPROVED** [review]
Tried this on archer2 and got the example running successfully. Came up with a couple of improvements but those can be done in separate PRs. Opened issues #46 and #47.

**Pull request merged**